src/main/java/com/google/codeu/servlets/AboutMeServlet.java

```diff
@@ -0,0 +1,65 @@
+package com.google.codeu.servlets;
+
+import java.io.IOException;
+
+import javax.servlet.annotation.WebServlet;
+import javax.servlet.http.HttpServlet;
+import javax.servlet.http.HttpServletRequest;
+import javax.servlet.http.HttpServletResponse;
+
+import com.google.appengine.api.users.UserService;
+import com.google.appengine.api.users.UserServiceFactory;
+import com.google.codeu.data.Datastore;
+
+/**
+ * Handles fetching and saving user data.
+ */
+@WebServlet("/about")
+public class AboutMeServlet extends HttpServlet {
+
+    private Datastore datastore;
+
+    @Override
+    public void init() {
+        datastore = new Datastore();
+    }
+
+    /**
+     * Responds with the "about me" section for a particular user.
+     */
+    @Override
+    public void doGet(HttpServletRequest request, HttpServletResponse response)
+            throws IOException {
+
+        response.setContentType("text/html");
+
+        String user = request.getParameter("user");
+
+        if(user == null || user.equals("")) {
+            // Request is invalid, return empty response
+            return;
+        }
+
+        String aboutMe = "This is " + user + "'s about me.";
+
+        response.getOutputStream().println(aboutMe);
+    }
+
+    @Override
+    public void doPost(HttpServletRequest request, HttpServletResponse response)
+            throws IOException {
+
+        UserService userService = UserServiceFactory.getUserService();
+        if (!userService.isUserLoggedIn()) {
```

Comment:
You can simplify this slightly by using a security constraint in the `web.xml` file:
https://cloud.google.com/appengine/docs/standard/java/config/webxml#security-auth

... to make App Engine automatically require the user to be logged-in.

Reply:
I believe that this will force the users to use gmail to login, is this something that we want to do?

Reply:
Re @markser 's feedback:

In App Engine, it is definitely easiest to just require a Google login as there is built-in support for that sort of thing. If you were creating an application where you wanted to support multiple different login options (e.g. Google, Facebook, Email+Password, etc.), you would probably want to use a solution like "Firebase Auth" (Google-owned) or "Auth0" (3rd party) to implement it (implementing authentication from scratch is not an easy task and probably not something that you ever want to do yourself, especially not without appropriate security people to vet it). In such a case, though, the code wouldn't look that dissimilar...

If you want to make this somewhat more realistic in terms of an application that supports multiple  kinds of authentication, then you could create a layer of indirection where you have your own set of authentication interfaces that wrap the App Engine auth libraries and that, in the future, could be replaced with an implementation that uses Firebase Auth or Auth0. 

For now, though, I'd recommend just learning how to use App Engine's google auth as it will provide you with an easier path to understanding these auth systems in general, and I don't think it is necessary to introduce all that complexity right away, especially if we don't ever plan to do the full blown thing in the course of this Code U (unless you want to make supporting multiple authentication mechanisms the main project that you work on down the road).